extensions/amp-story/1.0/amp-story.js

```diff
@@ -1632,6 +1635,10 @@ export class AmpStory extends AMP.BaseElement {
         actions.execute(this.sidebar_, 'open', /* args */ null,
             /* source */ null, /* caller */ null, /* event */ null,
             ActionTrust.HIGH);
+      } else if (this.sidebar_
+                && !sidebarState
+                && !this.maskElement_.hasAttribute('hidden')) {
+        this.closeOpacityMask_();
```

Comment:
Isn't this going to prevent from disconnecting the mutations observer (`else` statement right after)?
Do you think it'd be possible to do this from the code you just wrote L1626-1628?

Reply:
So, I think following the logic the mutations observer will still disconnect when the sidebar itself doesn't exist which is what we'd like. 

If I understand correctly, since we are trying to capture the moment when the sidebar is toggled from the visible to hidden state, we would need to dispatch a false here so that the sidebar is not toggled back on again, but we also don't necessary want to disconnect. 

I don't think that this can happen in L1626-1628 because then the mask will never be toggled on in the first place (since the sidebar is always hidden before the user opens the sidebar either through an element with an action or the hamburger menu).